internal/analyzers/prometheus.go

```diff
@@ -0,0 +1,292 @@
+			}
+			if len(nilNamespaceSelectors) == len(selectorLists) {
+				fmt.Println("All namespace selectors are either empty or properly defined.")
+			}
+		}
+	}
+
+	serviceSelectors := map[string]interface{}{
+		"ServiceMonitorSelector": prometheus.Spec.ServiceMonitorSelector,
+		"PodMonitorSelector":     prometheus.Spec.PodMonitorSelector,
+		"ProbeSelector":          prometheus.Spec.ProbeSelector,
+		"ScrapeConfigSelector":   prometheus.Spec.ScrapeConfigSelector,
+	}
+
+	serviceSelectorStatus, emptyServiceSelectors := checkPrometheusServiceSelectorsStatus(serviceSelectors)
+
+	if !serviceSelectorStatus {
+		return fmt.Errorf("No Service Selectors are defined, the Prometheus matches no objects, configuration is unmanaged")
+	}
+
+	if serviceSelectorStatus && emptyServiceSelectors!= nil {
+		checkServiceSelectorsCreated, svcselectorsLists := checkServiceDiscovery(ctx, clientSets, namespace)
+		if !checkServiceSelectorsCreated {
+			return fmt.Errorf("No Service Selectors are created yet")
+		}
+		if len(emptyServiceSelectors) == len(svcselectorsLists) {
+			fmt.Println("Prometheus matches all objects.")
+		}
+	}
+
+	slog.Info("Prometheus is compliant, no issues found", "name", name, "namespace", namespace)
+	return nil
+}
+
+func checkClusterRoleRules(crb v1.ClusterRoleBinding, cr *v1.ClusterRole) error {
+	var errs []string
+	verbsToCheck := []string{"get", "list", "watch"}
+	missingVerbs := []string{}
+
+	for _, rule := range cr.Rules {
+		for _, resource := range rule.Resources {
+			hasGet := false
+			if resource == "configmaps" {
+				for _, verb := range rule.Verbs {
+					if verb == "get" {
+						hasGet = true
+						break
+					}
+				}
+				if !hasGet {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'configmaps' with 'get' in its verbs", crb.RoleRef.Name))
+				}
+			} else {
+				for range rule.APIGroups {
+					for _, requiredVerb := range verbsToCheck {
+						found := false
+						for _, verb := range rule.Verbs {
+							if verb == requiredVerb {
+								found = true
+								break
+							}
+						}
+						if !found {
+							missingVerbs = append(missingVerbs, requiredVerb)
+						}
+					}
+					if len(missingVerbs) > 0 {
+						errs = append(errs, fmt.Sprintf("ClusterRole %s is missing necessary verbs for APIGroups: %v", crb.RoleRef.Name, missingVerbs))
+					}
+				}
+			}
+		}
+		for _, nonResource := range rule.NonResourceURLs {
+			if nonResource == "/metrics" {
+				hasGet := false
+				for _, verb := range rule.Verbs {
+					if verb == "get" {
+						hasGet = true
+						break
+					}
+				}
+				if !hasGet {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'get' verb for NonResourceURL '/metrics'", crb.RoleRef.Name))
+				}
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("Multiple errors found:\n%s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func checkPrometheusNamespaceSelectorsStatus(namespaceSelectors map[string]interface{}) (bool, []string) {
```

Comment:
I believe we can simplify this function a little bit.

Maybe we can have one function with the following signture, like this:
```
checkMonitorNamespaceSelectors(ctx context.Context, clientSets *k8sutil.ClientSets, labelSelector *metav1.LabelSelector) error
```

Basically this function will take a label selector and check if the namespace exists for given labelSelector.

This function usage would be something like this:

```
if err := checkMonitorNamespaceSelectors(ctx, prometheus.Spec.PodMonitorNamespaceSelector); err != nil {
		return fmt.Errorf("podMonitorNamespaceSelector is not properly defined: %s", err)
	}
```

wdyt?

Reply:
Added the change to the namespace selectors, pls review - doing something similar for the service selectors

Reply:
Also updated the service selectors, pls review